src/Miunie.Discord/Services/HelpService.cs

```diff
@@ -0,0 +1,129 @@
+﻿// This file is part of Miunie.
+//
+//  Miunie is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  Miunie is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with Miunie. If not, see <https://www.gnu.org/licenses/>.
+
+using Discord.Commands;
+using Discord.WebSocket;
+using Miunie.Core.Entities;
+using Miunie.Core.Providers;
+using Miunie.Discord.Attributes;
+using Miunie.Discord.Embeds;
+using Miunie.Discord.Entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Miunie.Discord
+{
+    public class HelpService
+    {
+        private readonly CommandService _commandService;
+        private readonly ILanguageProvider _lang;
+
+        public HelpService(CommandService commandService, ILanguageProvider lang)
+        {
+            _commandService = commandService;
+            _lang = lang;
+        }
+
+        public HelpResult GetDefault()
+        {
+            HelpResult entry = new HelpResult
+            {
+                Title = _lang.GetPhrase(PhraseKey.USER_EMBED_HELP_TITLE.ToString())
+            };
+
+            foreach (ModuleInfo module in _commandService.Modules)
+            {
+                string title = module.Name;
+                string content = string.Join(" ", module.Commands
+                    .GroupBy(x => x.Name)
+                    .Select(x => $"`{x.Key}`"));
+
+                entry.Sections.Add(new HelpSection(title, content));
+            }
```

Comment:
It would be much better to keep `entry.Sections` as `IEnumerable`, and extract the entire `foreach` into a well named method such as `GetAllModuleHelpSections()`.

That way this whole method would become trivial with:

```cs
return new HelpResult
{
    Title = _lang.GetPhrase(PhraseKey.USER_EMBED_HELP_TITLE.ToString()),
    Sections = GetAllModuleHelpSections()
};
```

> ⚠️ and of course use implicit declaration where possible...